src/google/protobuf/compiler/objectivec/objectivec_helpers.cc

```diff
@@ -192,15 +194,20 @@ const char* const kReservedWordList[] = {
     // http://publib.boulder.ibm.com/infocenter/lnxpcomp/v8v101/index.jsp?topic=%2Fcom.ibm.xlcpp8l.doc%2Flanguage%2Fref%2Fkeyw.htm
     "restrict",
 
+    // GCC/Clang extension
+    "typeof",
+
+    // Not a keyword, but will break you
+    "NULL",
+
     // Objective-C Runtime typedefs
     // From <obc/runtime.h>
     "Category", "Ivar", "Method", "Protocol",
 
     // NSObject Methods
-    // new is covered by C++ keywords.
-    "description", "debugDescription", "finalize", "hash", "dealloc", "init",
-    "class", "superclass", "retain", "release", "autorelease", "retainCount",
-    "zone", "isProxy", "copy", "mutableCopy", "classForCoder",
+    // These are brought in from an external header that is generated using
+    // method_dump.sh.
+#include "objectivec_nsobject_methods.h"
```

Comment:
This is likely going to be fragile for different build systems.  It it likely safer to have the header generate a real structures and then include it the same way all the other headers are imported in this file.

The update the code using this existing list to feel this local list and this external one into the single set for checking.

Reply:
Done